lib/salus/scanners/osv/base.rb

```diff
@@ -0,0 +1,148 @@
+require 'salus/scanners/base'
+require 'json'
+require 'zip'
+require 'uri'
+require 'net/http'
+
+module Salus::Scanners::OSV
+  class Base < Salus::Scanners::Base
+    DATABASE_STRING_MAPPING = {
+      "GHSA" => "Github Advisory Database",
+      "PYSEC" => "Python Packaging Advisory Database",
+      "GO" => "Go Vulnerability Database",
+      "RUSTSEC" => "RustSec Advisory Database",
+      "default" => "Open Source Vulnerabilitiy"
+    }.freeze
+
+    def run
+      raise NoMethodError, 'implement in subclass'
+    end
+
+    def should_run?
+      raise NoMethodError, 'implement in subclass'
+    end
+
+    def name
+      self.class.name.sub('Salus::Scanners::OSV::', '')
+    end
+
+    private
+
+    def osv_vulnerabilities
+      @osv_vulnerabilities ||= fetch_vulnerabilities
+    end
+
+    def osv_urls
+      urls = []
+      urls.append(osv_url_for("Go")) if @repository.go_sum_present? || @repository.go_mod_present?
+      urls.append(osv_url_for("PyPI")) if @repository.requirements_txt_present?
+      urls.append(osv_url_for("Maven")) if @repository.pom_xml_present? ||
+        @repository.build_gradle_present?
+
+      urls
+    end
+
+    def osv_url_for(package)
```

Comment:
At those sizes we should probably spin up a caching layer (given the volume of daily scans).

Let's create create a plugin hook here so we can easily override the loading of vulnerabilities as well as the matching of vulnerabilities.   This will allow salus to ship with a direct call (like you have here) and provide an easy override to cache locally.    Medium term we will plan on a normalized data layer for Salus - but we can tackle that w/out blocking this PR.

Reply:
Tracking internally.